pkgs/applications/networking/browsers/brave/default.nix

```diff
@@ -0,0 +1,108 @@
+{ stdenv, lib, fetchurl,
+  dpkg,
+  alsaLib,
+  at-spi2-atk,
+  atk,
+  cairo,
+  cups,
+  dbus,
+  expat,
+  fontconfig,
+  freetype,
+  gdk_pixbuf,
+  glib,
+  gnome2,
+  gtk3,
+  libuuid,
+  libX11,
+  libXcomposite,
+  libXcursor,
+  libXdamage,
+  libXext,
+  libXfixes,
+  libXi,
+  libXrandr,
+  libXrender,
+  libXScrnSaver,
+  libXtst,
+  nspr,
+  nss,
+  pango,
+  udev,
+  xorg,
+  zlib
+}:
+
+let rpath = lib.makeLibraryPath [
+    alsaLib
+    at-spi2-atk
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    fontconfig
+    freetype
+    gdk_pixbuf
+    glib
+    gnome2.GConf
+    gtk3
+    libuuid
+    libX11
+    libXcomposite
+    libXcursor
+    libXdamage
+    libXext
+    libXfixes
+    libXi
+    libXrandr
+    libXrender
+    libXScrnSaver
+    libXtst
+    nspr
+    nss
+    pango
+    udev
+    xorg.libxcb
+    zlib
+];
+
+
+in stdenv.mkDerivation rec {
+    name = "brave";
+    version = "0.24.0";
+
+    src = fetchurl {
+        url = "https://github.com/brave/browser-laptop/releases/download/v${version}dev/brave_${version}_amd64.deb";
```

Comment:
The github page redirects me to this site: https://github.com/brave/brave-browser

Reply:
brave-browser is their next "release", but at this moment the README still says: 

````
Downloads
We're not offering downloads yet for this next generation desktop browser.
````

I'm also not sure about what github page do you mean? I tried the URL and successfully downloaded the package?

````
$ wget https://github.com/brave/browser-laptop/releases/download/v0.24.0dev/brave_0.24.0_amd64.deb
--2018-10-06 20:51:29--  https://github.com/brave/browser-laptop/releases/download/v0.24.0dev/brave_0.24.0_amd64.deb
Resolving github.com (github.com)... 192.30.253.112, 192.30.253.113
Connecting to github.com (github.com)|192.30.253.112|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/46891997/60b47c00-bb32-11e8-9797-9f9575ff9a2c?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181006%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181006T185129Z&X-Amz-Expires=300&X-Amz-Signature=b1803354e98ab5e2b820fce5ff0bb13779c8c445774a5e55a4ea82189c4062b2&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dbrave_0.24.0_amd64.deb&response-content-type=application%2Foctet-stream [following]
--2018-10-06 20:51:29--  https://github-production-release-asset-2e65be.s3.amazonaws.com/46891997/60b47c00-bb32-11e8-9797-9f9575ff9a2c?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181006%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181006T185129Z&X-Amz-Expires=300&X-Amz-Signature=b1803354e98ab5e2b820fce5ff0bb13779c8c445774a5e55a4ea82189c4062b2&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dbrave_0.24.0_amd64.deb&response-content-type=application%2Foctet-stream
Resolving github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.216.105.83
Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.216.105.83|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 97726210 (93M) [application/octet-stream]
Saving to: ‘brave_0.24.0_amd64.deb’

brave_0.24.0_amd64.deb                                    100%[===================================================================================================================================>]  93.20M  1.78MB/s    in 53s     

2018-10-06 20:52:23 (1.76 MB/s) - ‘brave_0.24.0_amd64.deb’ saved [97726210/97726210]
````